### PR TITLE
Ignore paths with invalid chars in PathWhich

### LINKED ISSUE #39432: modules/mono/mono_gd/gd_mono_utils.cpp:371 - System.ArgumentException: Illegal characters in path.
**Godot version:**
v3.2.2.beta4.mono.official

**OS/device including version:**

Windows 10 Pro 1909

**Issue description:**

When opening a project, in the editor output, I get flooded with about 20 of the following errors:
`modules/mono/mono_gd/gd_mono_utils.cpp:371 - System.ArgumentException: Illegal characters in path`
These then change to:
`modules/mono/mono_gd/gd_mono_utils.cpp:371 - System.NullReferenceException: Object reference not set to an instance of an object`
which continue as long as the editor is open.

These are expanded in the console as:
```
Unhandled Exception:
System.ArgumentException: Illegal characters in path.
  at System.IO.Path.Combine (System.String path1, System.String path2) [0x0003e] in <34e7ab706f1141f9b981d74fbe2f5560>:0
  at GodotTools.Utils.OS+<>c__DisplayClass39_0.<PathWhichWindows>b__1 (System.String dir) [0x00000] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
  at System.Linq.Enumerable+SelectListIterator`2[TSource,TResult].MoveNext () [0x00048] in <fdc50bfd761741c0a0c2f052bf074a1c>:0
  at System.Linq.Enumerable+<SelectManyIterator>d__167`3[TSource,TCollection,TResult].MoveNext () [0x000d1] in <fdc50bfd761741c0a0c2f052bf074a1c>:0
  at System.Linq.Enumerable.TryGetFirst[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate, System.Boolean& found) [0x0004f] in <fdc50bfd761741c0a0c2f052bf074a1c>:0
  at System.Linq.Enumerable.FirstOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source, System.Func`2[T,TResult] predicate) [0x00000] in <fdc50bfd761741c0a0c2f052bf074a1c>:0
  at GodotTools.Utils.OS.PathWhichWindows (System.String name) [0x00111] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
  at GodotTools.Utils.OS.PathWhich (System.String name) [0x0000e] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
  at GodotTools.BuildManager.Initialize () [0x00023] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
  at GodotTools.GodotSharpEditor.EnablePlugin () [0x00453] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
ERROR: debug_send_unhandled_exception_error: System.ArgumentException: Illegal characters in path.
   At: modules/mono/mono_gd/gd_mono_utils.cpp:371
```
and
```
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at GodotTools.HotReloadAssemblyWatcher.RestartTimer () [0x00000] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
  at GodotTools.HotReloadAssemblyWatcher._Notification (System.Int32 what) [0x00008] in <bf243baa94fe41ccbd98c8bbc3a78900>:0
  at (wrapper managed-to-native) System.Threading.Monitor.Exit(object)
  at Godot.GodotTaskScheduler.ExecuteQueuedTasks () [0x00050] in <e870abd78d2142af92585d192bb3eb0a>:0
  at Godot.GodotTaskScheduler.Activate () [0x00001] in <e870abd78d2142af92585d192bb3eb0a>:0
  at (wrapper native-to-managed) Godot.GodotTaskScheduler.Activate(Godot.GodotTaskScheduler,System.Exception&)
ERROR: debug_send_unhandled_exception_error: System.NullReferenceException: Object reference not set to an instance of an object
   At: modules/mono/mono_gd/gd_mono_utils.cpp:371
```
respectively.

**Steps to reproduce:**
- Run [Godot 3.2.2 beta4 mono win64](https://downloads.tuxfamily.org/godotengine/3.2.2/beta4/mono/Godot_v3.2.2-beta4_mono_win64.zip).
- Create a new project. I used a new folder: `C:\Users\Marcel\Downloads\Godot\Test Mono Beta 4`

**Minimal reproduction project:**
None required.

### PR DESCRIPTION
**Pull request by raulsntos**
Fixes #39432

### CODE REVIEW

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs [hunk 1]
```diff
@@ -113,11 +113,18 @@ private static string PathWhichWindows([NotNull] string name)
...
+            char[] invalidPathChars = Path.GetInvalidPathChars();
 
             var searchDirs = new List<string>();
 
             if (pathDirs != null)
-                searchDirs.AddRange(pathDirs);
+            {
+                foreach (var pathDir in pathDirs)
+                {
+                    if (pathDir.IndexOfAny(invalidPathChars) != -1) continue;
```
Comment (neikeq) [inline modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs:124]: The continue should go into its own line.

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs [hunk 2]
```diff
@@ -137,11 +144,18 @@ into path
...
```
Comment (neikeq) [inline modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs:155]: ditto

**Review by neikeq — CHANGES_REQUESTED** [review]
Looks good, just need code style fixes.

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs (hunk 1 above)
Comment (raulsntos) [reply modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs]: Fixed

File: modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs (hunk 2 above)
Comment (raulsntos) [reply modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs]: Fixed

**Review by neikeq — APPROVED** [review]

**Comment by neikeq** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.3.

**Pull request merged**